### PR TITLE
Lp profit

### PR DESCRIPTION
**Pull request by wakamex**

### CODE REVIEW

**Comment by wakamex** [comment]
lgtm

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -17,8 +21,9 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          python -m pip install black~=22.0 pylint numpy pandas
+          python -m pip install black~=22.0 pylint
+          python -m pip install -r requirements.txt
       - name: Autolinting code with black
-        run: black $(git ls-files '*.py')
+        run: black $(git ls-files '*.py') --preview -l 120 -t py311
```
Comment (dpaiton) [inline .github/workflows/lint.yml:27]: What are these changes for? The line number should be set in the config file already. And the target python version should be set already in the matrix strategy.

File: examples/fee_analysis.ipynb [hunk 2]
```diff
@@ -52,6 +52,8 @@
         "    \"max_target_volume\": 0.01, # fration of pool liquidity\n",
         "    \"min_pool_apy\": 0.02, # as a decimal\n",
         "    \"max_pool_apy\": 0.9, # as a decimal\n",
+        "    \"pool_apy_target_range\": [0.15,0.20], # as a decimal\n",
```
Comment (dpaiton) [inline examples/fee_analysis.ipynb:55]: You got rid of this elsewhere, correct?

File: examples/fee_analysis.ipynb [hunk 3]
```diff
@@ -119,7 +122,8 @@
       "metadata": {},
       "outputs": [],
       "source": [
-        "[trades, trades_agg] = format_trades(simulator.analysis_dict)"
+        "[trades, trades_agg] = format_trades(simulator.analysis_dict)\n",
+        "# display(trades.head(1).T)"
```
Comment (dpaiton) [inline examples/fee_analysis.ipynb:126]: delete

File: examples/fee_analysis.ipynb [hunk 4]
```diff
@@ -279,52 +290,144 @@
         "ax[current_plot].xaxis.set_ticklabels([])\n",
         "ax[current_plot].legend(fontsize=18)\n",
         "\n",
-        "# price per share & spot price\n",
+        "### direction parameters ###\n",
```
Comment (dpaiton) [inline examples/fee_analysis.ipynb:293]: Was this for Will's latest plot? If so, can this stuff go into a new notebook?

File: pyproject.toml [hunk 5]
```diff
@@ -30,4 +30,8 @@ addopts = [
 ]
 
 [tool.pylint.format]
-max-line-length = "120"
\ No newline at end of file
+max-line-length = "120"
+
+[tool.black]
+line-length = 120
+target-version = ['py311']
```
Comment (dpaiton) [inline pyproject.toml:37]: Let's let the linter config matrix determine target-version

File: examples/fee_analysis.ipynb (hunk 2 above)
Comment (wakamex) [reply examples/fee_analysis.ipynb:55]: yes shouldn't be in there

File: src/elfpy/simulators.py [hunk 6]
```diff
@@ -300,9 +282,10 @@ def run_simulation(self, override_dict=None):
             # Could define a 'trade amount & direction' time series that's a function of the vault apy
             # Could support using actual historical trades or a fit to historical trades)
             day_trading_volume = 0
+            days_trades = []
             while day_trading_volume < self.target_daily_volume:
-                # Compute tokens to swap
                 token_index = self.rng.integers(low=0, high=2)  # 0 or 1
+                days_trades.append(token_index)
```
Comment (dpaiton) [inline src/elfpy/simulators.py:288]: This is not used anywhere -- is it necessary?

File: src/elfpy/utils/data.py [hunk 7]
```diff
@@ -18,23 +18,45 @@ def format_trades(analysis_dict):
...
-    token_asset_liquidity_usd = (
-        trades.token_asset_reserves * trades.base_asset_price * trades.spot_price
-    )
+    token_asset_liquidity_usd = trades.token_asset_reserves * trades.base_asset_price * trades.spot_price
     trades["total_liquidity_usd"] = base_asset_liquidity_usd + token_asset_liquidity_usd
     trades["trade_volume_usd"] = trades.out_with_fee
     # calculate percent change in spot price since the first spot price (after first trade, kinda weird)
-    trades["price_rate_of_change"] = (
+    trades["price_total_return"] = (
         trades.loc[:, "spot_price"] / trades.loc[0, "spot_price"] - 1
```
Comment (dpaiton) [inline src/elfpy/utils/data.py:26]: Did you verify that all of the example notebooks still run with this change?

File: src/elfpy/utils/data.py [hunk 8]
```diff
@@ -18,23 +18,45 @@ def format_trades(analysis_dict):
...
+    trades["share_price_total_return_percent_annualized"]  = scale * trades["share_price_total_return_percent"]
+
     # create explicit column that increments per trade
     trades = trades.reset_index()
-    # aggregate trades over days
+
+    ### STATS AGGREGATED BY DAY ###
     keep_columns = [
-        "model_name",
+        "model_name" if trades.model_name.nunique() > trades.scenario_name.nunique() else "scenario_name",
```
Comment (dpaiton) [inline src/elfpy/utils/data.py:59]: Can you add a comment noting the difference between "model_name" and "scenario_name"?

File: src/elfpy/utils/data.py [hunk 9]
```diff
@@ -48,8 +70,9 @@ def format_trades(analysis_dict):
...
-    for model in trades_agg.model_name.unique():
-        trades_agg.loc[
-            trades_agg.model_name == model, "fee_in_usd_cum_sum"
-        ] = trades_agg.loc[trades_agg.model_name == model, "fee_in_usd_sum"].cumsum()
+    for model in trades_agg.loc[:,keep_columns[0]].unique():
+        trades_agg.loc[trades_agg.loc[:,keep_columns[0]] == model, "fee_in_usd_cum_sum"] = trades_agg.loc[
+            trades_agg.loc[:,keep_columns[0]] == model, "fee_in_usd_sum"
+        ].cumsum()
     return [trades, trades_agg]
+
```
Comment (dpaiton) [inline src/elfpy/utils/data.py:78]: add newline back in

File: examples/fee_analysis.ipynb (hunk 4 above)
Comment (wakamex) [reply examples/fee_analysis.ipynb:291]: good catch yes... removing them there

will's charts are in there is a separate notebook: examples/rate_scenarios.ipynb

File: src/elfpy/utils/data.py (hunk 7 above)
Comment (wakamex) [reply src/elfpy/utils/data.py:26]: definitely not, but fee_analysis and rate_scenario do. i'll check now.

Comment (wakamex) [reply src/elfpy/utils/data.py:26]: ctrl-shift-f found 5 lines i hadn't manually changed. done now.

**Review by dpaiton — APPROVED** [review]

File: src/elfpy/utils/data.py (hunk 9 above)
Comment (wakamex) [reply src/elfpy/utils/data.py:83]: you're trolling me, right?

**Pull request merged**